### ISSUE TITLE
Bug in opt -full

### ISSUE THREAD

**Issue by mtuomi** [cliffordwolf/yosys#306]
```
module ltbug (input [3:0] a, output res);
    assign res = a < 6'b100000;
endmodule
```

Above code crashes yosys with opt -full command.

The crash happens in opt_expr.cc on this line : 
`					RTLIL::SigSpec a_prime(RTLIL::State::S0, width - bit_set);
`
as width - bit_set becomes negative.

**Comment by C-Elegans** [comment]
I should have a fix out by tomorrow (I wrote that part).

**Comment by cliffordwolf** [comment]
Fixed by #308.
